parser/src/main/scala/aqua/parser/expr/ArrowTypeExpr.scala

```diff
@@ -19,8 +20,9 @@ object ArrowTypeExpr extends Expr.Leaf {
 
   override val p: Parser[ArrowTypeExpr[Span.S]] =
     (Name.p ~ ((` : ` *> ArrowTypeToken.`arrowdef`(
-      DataTypeToken.`datatypedef`
-    )) | ArrowTypeToken.`arrowWithNames`(DataTypeToken.`datatypedef`))).map { case (name, t) =>
-      ArrowTypeExpr(name, t)
+      CompositeTypeToken.`compositetypedef`
```

Comment:
Maybe rename `BasicTypeToken -> ScalarTypeToken` and `CompositeTypeToken -> BasicTypeToken`?

Reply:
also, it is outdated, that `ArrowTypeToken` is not a `CompositeTypeToken`. Should we put everything just under `TypeToken`?